tools/terraform_sync_tool/terraform_sync.py

```diff
@@ -0,0 +1,84 @@
+    return table_schemas
+
+# Identify tables with drifts and add return tables_of_interest 
+def get_drifted_tables(json_file):
+    tables_of_interest = {} # store events(dict) of interest where key=resource_name+resource_key as identifier and value=resource_id                                                                                                    
+    # Opening JSON file
+    with open(json_file) as file:
+        lines = file.readlines()
+        # Scan json lines to store events of interest
+        for line in reversed(lines):                                                                    
+            json_line = json.loads(line)
+            type = json_line.get('type')
+            change = json_line.get('change')   
+            hook = json_line.get('hook') 
+            # When resource_drift event detected, append new dict to tables_of_interest
+            if type == 'resource_drift' and change and change.get('resource') and change.get('resource').get('resource_type') == 'google_bigquery_table':
+                change_resource = change.get('resource')
+                # Use condensed resource_name+resource_key as key of new dict
+                resource_condensed = change_resource.get('resource_name')+change_resource.get('resource_key')
+                # Use resource infomation from resource_drift event in value of new dict
+                resource_from_drift = {'identifier':
+                    {
+                    'resource_name':change_resource.get('resource_name'),
+                    'resource_key':change_resource.get('resource_key')
+                    }}
+                # Add to tables_of_interest
+                tables_of_interest[resource_condensed] =  resource_from_drift
+            # When refresh_complete event detected, add id_value and convert it to fully-qualified table_id, and add table_id to dict
+            if type == 'refresh_complete' and hook and hook.get('resource') and hook.get('resource').get('resource_type') == 'google_bigquery_table':
+                hook_resource = hook.get('resource')
+                # Use condensed resource_name+resource_key as the key to check if it exists in tables_of_interest
+                resource_condensed = hook_resource.get('resource_name') + hook_resource.get('resource_key')
+                # Usee resource information from refresh_complete event as identifier
+                resource_from_hook = {
+                        'resource_name':hook_resource.get('resource_name'),
+                        'resource_key':hook_resource.get('resource_key')
+                        }
+                # When the resource_condensed exists and resource_from_hook matched with identifier in dict
+                # add id_value and convert it to fully-qualified table_id, and add table_id to dict
+                if resource_condensed in tables_of_interest and resource_from_hook == tables_of_interest.get(resource_condensed).get('identifier'):
```

Comment:
You're checking the same thing twice. resource_condensed is made up of resource_name and resource_key. The value of .get('identifier') is the resource_name and resource_key

Reply:
Updated in https://github.com/GoogleCloudPlatform/bigquery-utils/pull/290/commits/69a2df35cf0d995b0ca94c95a2a5fbc79aeaeb58